### PR TITLE
Add pipelines and required modules of QwenImage in Diffusers Master

### PR DESCRIPTION
**Pull request by Dong1017**
# What does this PR do? 
 
 
# Adds 
## 1. QwenImage Pipelines and Required Modules  
*(Comparable with Diffusers Master)*

### a. Pipelines
- `mindone.diffusers.QwenImagePipeline`
- `mindone.diffusers.QwenImageImg2ImgPipeline`
- `mindone.diffusers.QwenImageInpaintPipeline`
- `mindone.diffusers.QwenImageEditPipeline`
- `mindone.diffusers.QwenImageEditInpaintPipeline`

### b. Modules
- `mindone.diffusers.models.AutoencoderQwenImage`
- `mindone.diffusers.models.QwenImageTransformer2DModel`
- `mindone.diffusers.loaders.QwenImageLoraLoaderMixin`

## 2. add UTs of pipelines — all passed
*(targeting Diffusers Master)*

- `tests/diffusers_tests/pipelines/qwenimage/test_qwenimage.py`
- `tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_img2img.py`
- `tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_inpaint.py`
- `tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_edit.py`

# Usage 
- QwenImagePipeline 
```py 
import mindspore as ms 
from mindone.diffusers import QwenImagePipeline 

pipe = QwenImagePipeline.from_pretrained("Qwen/Qwen-Image", mindspore_dtype=ms.bfloat16) 
prompt = "A cat holding a sign that says hello world" 
# Depending on the variant being used, the pipeline call will slightly vary. 
# Refer to the pipeline documentation for more details. 
image = pipe(prompt, num_inference_steps=50)[0][0] 
image.save("qwenimage.png") 
``` 
- QwenImageImg2ImgPipeline 
```py 
import mindspore as ms 
from mindone.diffusers import QwenImageImg2ImgPipeline
from mindone.diffusers.utils import load_image

pipe = QwenImageImg2ImgPipeline.from_pretrained("Qwen/Qwen-Image", mindspore_dtype=mindspore.bfloat16)
url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"
init_image = load_image(url).resize((1024, 1024))
prompt = "cat wizard, gandalf, lord of the rings, detailed, fantasy, cute, adorable, Pixar, Disney"
images = pipe(prompt=prompt, negative_prompt=" ", image=init_image, strength=0.95)[0][0]
images.save("qwenimage_img2img.png")
``` 
- QwenImageInpaintPipeline 
```py 
import mindspore as ms 
from mindone.diffusers import QwenImageInpaintPipeline 
from mindone.diffusers.utils import load_image 

pipe = QwenImageInpaintPipeline.from_pretrained("Qwen/Qwen-Image", mindspore_dtype=ms.bfloat16) 
prompt = "Face of a yellow cat, high resolution, sitting on a park bench" 
img_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo.png" 
mask_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo_mask.png" 
source = load_image(img_url) 
mask = load_image(mask_url) 
image = pipe(prompt=prompt, negative_prompt=" ", image=source, mask_image=mask, strength=0.85)[0][0] 
image.save("qwenimage_inpainting.png") 
``` 
- QwenImageEditPipeline 
```py 
import mindspore as ms 
from PIL import Image 
from mindone.diffusers import QwenImageEditPipeline 
from mindone.diffusers.utils import load_image 

pipe = QwenImageEditPipeline.from_pretrained("Qwen/Qwen-Image-Edit", mindspore_dtype=ms.bfloat16) 
image = load_image("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/yarn-art-pikachu.png").convert("RGB") 
prompt = ("Make Pikachu hold a sign that says 'Qwen Edit is awesome', yarn art style, detailed, vibrant colors") 
# Depending on the variant being used, the pipeline call will slightly vary. 
# Refer to the pipeline documentation for more details. 
image = pipe(image, prompt, num_inference_steps=50)[0][0] 
image.save("qwenimage_edit.png") 
``` 
- QwenImageEditInpaintPipeline 
```py 
import mindspore as ms 
from PIL import Image
from mindone.diffusers import QwenImageEditInpaintPipeline
from mindone.diffusers.utils import load_image

pipe = QwenImageEditInpaintPipeline.from_pretrained("Qwen/Qwen-Image-Edit", mindspore_dtype=mindspore.bfloat16)
prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
img_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo.png"
mask_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo_mask.png"
source = load_image(img_url)
mask = load_image(mask_url)
image = pipe(prompt=prompt, negative_prompt=" ", image=source, mask_image=mask, strength=1.0, num_inference_steps=50)[0][0]
image.save("qwenimage_inpainting.png")
``` 

# Performance 
*(Experiments are tested on Ascend Atlas 800T A2 machines with MindSpore 2.7.0, setting to pynative mode)*
| Pipeline | Speed |
|------|------|
| QwenImagePipeline | 9.93 s/it |
| QwenImageImg2ImgPipeline| 9.56 s/it |
| QwenImageInpaintPipeline| 4.80 s/it |
| QwenImageEditPipeline| 13.25 s/it |
| QwenImageEditInpaintPipeline| 13.98 s/it |

# Notes 
1. require transformers==4.52.1 
2. The produced pictures are nearly identical to those by Torch, when setting consistent `random seed` and `hidden states` from the text encoder.
3. TODO: jit mode; UTs of modules

## Before submitting 
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case). 
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)? 
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the 
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83) 
- [ ] Did you build and run the code without any errors? 
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks) 
- [ ] Did you write any new necessary tests? 
## Who can review? 
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag 
members/contributors who may be interested in your PR. 
@xxx

### CODE REVIEW

File: mindone/diffusers/loaders/lora_pipeline.py [hunk 1]
```diff
@@ -6029,3 +6030,333 @@ def __init__(self, *args, **kwargs):
...
+
+        Parameters:
+            pretrained_model_name_or_path_or_dict (`str` or `os.PathLike` or `dict`):
+                Can be either:
+
+                    - A string, the *model id* (for example `google/ddpm-celebahq-256`) of a pretrained model hosted on
+                      the Hub.
+                    - A path to a *directory* (for example `./my_model_directory`) containing the model weights saved
+                      with [`ModelMixin.save_pretrained`].
+                    - A [torch state
```
Comment (Cui-yshoho) [inline mindone/diffusers/loaders/lora_pipeline.py:6068]: mindspore state xx

File: mindone/diffusers/loaders/lora_pipeline.py [hunk 2]
```diff
@@ -6029,3 +6030,333 @@ def __init__(self, *args, **kwargs):
...
+                Controls how much to influence the outputs with the LoRA parameters.
+            safe_fusing (`bool`, defaults to `False`):
+                Whether to check fused weights for NaN values before fusing and if values are NaN not fusing them.
+            adapter_names (`List[str]`, *optional*):
+                Adapter names to be used for fusing. If nothing is passed, all active adapters will be fused.
+
+        Example:
+
+        ```py
+        from diffusers import DiffusionPipeline
```
Comment (Cui-yshoho) [inline mindone/diffusers/loaders/lora_pipeline.py:6328]: Example没有修改

File: mindone/diffusers/models/autoencoders/autoencoder_kl_qwenimage.py [hunk 3]
```diff
@@ -0,0 +1,1089 @@
...
+# - GitHub: https://github.com/Wan-Video/Wan2.1
+# - arXiv: https://arxiv.org/abs/2503.20314
+
+from typing import List, Optional, Tuple, Union
+import math
+import numpy as np
+
+import mindspore as ms
+from mindspore import mint, nn, ops
+# import torch
```
Comment (Cui-yshoho) [inline mindone/diffusers/models/autoencoders/autoencoder_kl_qwenimage.py:30]: 这些注释可以直接删掉 后面应该还有一些类似的 都可以删掉

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+    ):
+        dtype = dtype or self.text_encoder.dtype
+
+        prompt = [prompt] if isinstance(prompt, str) else prompt
+
+        template = self.prompt_template_encode
+        drop_idx = self.prompt_template_encode_start_idx
+        txt = [template.format(e) for e in prompt]
+        txt_tokens = self.tokenizer(
+            txt, max_length=self.tokenizer_max_length + drop_idx, padding=True, truncation=True, return_tensors="np" \
```
Comment (Cui-yshoho) [inline mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:195]: 这里格式看起来有点奇怪

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+            ""
+        )
+        encoder_hidden_states = self.text_encoder(
+            input_ids=ms.Tensor(txt_tokens.input_ids),
+            attention_mask=ms.Tensor(txt_tokens.attention_mask),
```
Comment (Cui-yshoho) [inline mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:200]: tensor使用优先ms.tensor 后面应该还有一些类似Tensor的问题

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+                will be used.
+            guidance_scale (`float`, *optional*, defaults to 3.5):
+                Guidance scale as defined in [Classifier-Free Diffusion
+                Guidance](https://huggingface.co/papers/2207.12598). `guidance_scale` is defined as `w` of equation 2.
+                of [Imagen Paper](https://huggingface.co/papers/2205.11487). Guidance scale is enabled by setting
+                `guidance_scale > 1`. Higher guidance scale encourages to generate images that are closely linked to
+                the text `prompt`, usually at the expense of lower image quality.
+            num_images_per_prompt (`int`, *optional*, defaults to 1):
+                The number of images to generate per prompt.
+            generator (`torch.Generator` or `List[torch.Generator]`, *optional*):
```
Comment (Cui-yshoho) [inline mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:470]: np.random.generator

File: tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_img2img.py [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+        ms_pipe = ms_pipe_cls(**ms_components)
+
+        pt_pipe.set_progress_bar_config(disable=None)
+        ms_pipe.set_progress_bar_config(disable=None)
+
+        ms_dtype, pt_dtype = getattr(ms, dtype), getattr(torch, dtype)
+        pt_pipe = pt_pipe.to(pt_dtype)
+        ms_pipe = ms_pipe.to(ms_dtype)
+
+        sys.modules[ms_pipe.__module__].randn_tensor = randn_tensor
```
Comment (Cui-yshoho) [inline tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_img2img.py:211]: 为什么这里要单独替换呀？

File: tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_edit.py [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+        
+        model_id = "Qwen/Qwen-Image-Edit"
+
+        pipe = QwenImageEditPipeline.from_pretrained(model_id, mindspore_dtype=ms_dtype)
+
+        pipe.vae.enable_tiling()
+
+        torch.manual_seed(0)
+        image = pipe(
+            image=Image.new("RGB", (32, 32)),
```
Comment (Cui-yshoho) [inline tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_edit.py:257]: 这里不需要load实际image吗？

**Comment by SamitHuang** [comment]
How to fix the requirement of `transformers==4.52.1`?

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+                ms.Tensor(self.vae.config.latents_mean)
+                .view(1, self.vae.config.z_dim, 1, 1, 1)
+                .to(latents.dtype)
+            )
+            latents_std = 1.0 / ms.Tensor(self.vae.config.latents_std).view(1, self.vae.config.z_dim, 1, 1, 1).to(
+                latents.dtype
+            )
+            latents = latents / latents_std + latents_mean
+            # TODO: we use pynative mode here since cache in vae.decode which not supported in graph mode
+            with pynative_context():
```
Comment (SamitHuang) [inline mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:685]: Is it necessary? since  the default MS mode and the supported mode is pynative?

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+            Variational Auto-Encoder (VAE) Model to encode and decode images to and from latent representations.
+        text_encoder ([`Qwen2.5-VL-7B-Instruct`]):
+            [Qwen2.5-VL-7B-Instruct](https://huggingface.co/Qwen/Qwen2.5-VL-7B-Instruct), specifically the
+            [Qwen2.5-VL-7B-Instruct](https://huggingface.co/Qwen/Qwen2.5-VL-7B-Instruct) variant.
+        tokenizer (`QwenTokenizer`):
+            Tokenizer of class
+            [CLIPTokenizer](https://huggingface.co/docs/transformers/en/model_doc/clip#transformers.CLIPTokenizer).
+    """
+
+    model_cpu_offload_seq = "text_encoder->transformer->vae"
```
Comment (SamitHuang) [inline mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:145]: Is the offloading seq supported?

**Comment by SamitHuang** [comment]
Can add an inference example and lora fine-tune example  in `examples` folder, which helps introduce QwenImage

File: tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_edit.py (hunk 8 above)
Comment (Dong1017) [reply tests/diffusers_tests/pipelines/qwenimage/test_qwenimage_edit.py:257]: 这里和hf仓ut保持一致

File: mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py (hunk 9 above)
Comment (Dong1017) [reply mindone/diffusers/pipelines/qwenimage/pipeline_qwenimage.py:685]: Thanks for the review. `with pynavtive_context()` will be removed, since the double-check shows that its impact on results is minimal. Tests for all pipelines are doing.

File: mindone/diffusers/loaders/lora_pipeline.py [hunk 11]
```diff
@@ -6029,3 +6030,333 @@ def __init__(self, *args, **kwargs):
...
+        """
+        # Load the layers corresponding to transformer.
+        logger.info(f"Loading {cls.transformer_name}.")
+        transformer.load_lora_adapter(
+            state_dict,
+            network_alphas=None,
+            adapter_name=adapter_name,
+            metadata=metadata,
+            _pipeline=_pipeline,
+            low_cpu_mem_usage=low_cpu_mem_usage,
```
[the remark this answers is not in the document]
Comment (Dong1017) [reply mindone/diffusers/loaders/lora_pipeline.py:6236]: 已删除`low_cpu_mem_usage`相关用法
